crates/rust-analyzer/src/bin/logger.rs

```diff
@@ -48,31 +48,31 @@ impl Log for Logger {
             return;
         }
 
-        let should_flush = match &self.file {
+        match &self.file {
             Some(w) => {
+                let mut writer = w.lock();
                 let _ = writeln!(
-                    w.lock(),
+                    writer,
                     "[{} {}] {}",
                     record.level(),
                     record.module_path().unwrap_or_default(),
                     record.args(),
                 );
-                self.no_buffering
+
+                if self.no_buffering {
+                    writer.flush().unwrap();
```

Comment:
Maybe

```suggestion
                    let _ = writer.flush();
```

But I'm not sure how I feel about it. This might cause some spurious panics when we're run by another program and it closes our `stderr`.

Reply:
Yeah, I only put the unwrap() to get rid of the warning. thanks for mentioning.

Reply:
Yeah, then maybe let's ignore the errors here.